testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiSparkCompatibility.java

```diff
@@ -0,0 +1,325 @@
+                        "   ts bigint\n" +
+                        ")\n" +
+                        "WITH (\n" +
+                        "   location = 's3://%s/%s'\n" +
+                        ")",
+                tableName,
+                bucketName,
+                tableDirectory);
+
+        try {
+            assertThat(onTrino().executeQuery("SHOW CREATE TABLE hudi.default." + tableName)).containsExactlyInOrder(row(showCreateTable));
+        }
+        finally {
+            onHudi().executeQuery("DROP TABLE default." + tableName);
+        }
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testCopyOnWriteTableSelect()
+    {
+        String tableName = "test_hudi_cow_select_" + randomTableSuffix();
+        String tableDirectory = "hudi-compatibility-test-" + tableName;
+
+        createNonPartitionedTable(tableName, tableDirectory, COW_TABLE_TYPE);
+
+        List<QueryAssert.Row> expectedRows = ImmutableList.of(
+                row(1, "a1"),
+                row(2, "a2"));
+
+        try {
+            assertThat(onHudi().executeQuery("SELECT id, name FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT id, name FROM hudi.default." + tableName))
+                    .containsOnly(expectedRows);
+        }
+        finally {
+            onHudi().executeQuery("DROP TABLE default." + tableName);
+        }
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testCopyOnWritePartitionedTableSelect()
+    {
+        String tableName = "test_hudi_cow_partitioned_select_" + randomTableSuffix();
+        String tableDirectory = "hudi-compatibility-test-" + tableName;
+
+        createPartitionedTable(tableName, tableDirectory, COW_TABLE_TYPE);
+
+        List<QueryAssert.Row> expectedRows = ImmutableList.of(
+                row(1, "a1", 1000, "2021-12-09", "10"),
+                row(2, "a2", 1000, "2021-12-09", "11"));
+
+        try {
+            assertThat(onHudi().executeQuery("SELECT id, name, ts, dt, hh FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT id, name, ts, dt, hh FROM hudi.default." + tableName))
+                    .containsOnly(expectedRows);
+
+            expectedRows = ImmutableList.of(row(2, "a2", 1000));
+            assertThat(onTrino().executeQuery("SELECT id, name, ts FROM hudi.default." + tableName + " WHERE dt = '2021-12-09' AND hh = '11'"))
+                    .containsOnly(expectedRows);
+        }
+        finally {
+            onHudi().executeQuery("DROP TABLE default." + tableName);
+        }
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testCopyOnWriteTableSelectAfterUpdate()
+    {
+        String tableName = "test_hudi_cow_select_after_update" + randomTableSuffix();
+        String tableDirectory = "hudi-compatibility-test-" + tableName;
```

Comment:
I see this variable definition in each test. Is it actually necessary?
I feel that the `tableName` is expressive enough.

Reply:
Yeah that's correct. It is not really necessary.